test/__snapshots__/test.js.snap

```diff
@@ -5248,6 +5402,130 @@ const Impl = require(\\"../implementations/URL.js\\");
 "
 `;
 
+exports[`with processors URLCallback.webidl 1`] = `
+"\\"use strict\\";
+
+const conversions = require(\\"webidl-conversions\\");
+const utils = require(\\"./utils.js\\");
+
+const URL = require(\\"./URL.js\\");
+
+exports.convert = (value, { context = \\"The provided value\\" } = {}) => {
+  if (typeof value !== \\"function\\") {
+    throw new TypeError(context + \\" is not a function\\");
+  }
+
+  function invokeTheCallbackFunction(...args) {
+    const thisArg = utils.tryWrapperForImpl(this);
+    let callResult;
+
+    for (let i = 0; i < Math.min(args.length, 2); i++) {
+      args[i] = utils.tryWrapperForImpl(args[i]);
+    }
+
+    if (args.length < 2) {
+      for (let i = args.length; i < 2; i++) {
+        args[i] = undefined;
+      }
+    } else if (args.length > 2) {
+      args.length = 2;
+    }
```

Comment:
All this args handling seems unnecessary. Can we assume that the caller (which will be jsdom code) will always pass the correct number of arguments, and then just do this similar to what callback interfaces did? So this ends up reducing to

```js
function invokeTheCallbackFunction(url, string) {
  // ...
  url = utils.tryWrapperForImpl(url);
  /* optional, if it's easier to not distinguish interfaces etc.: */ string = utils.tryWrapperForImpl(string);
  // ...
}
```

Reply:
The issue is that we have to support optional and variadic arguments in callback functions, because we support APIs that use them.

Reply:
Sure, the same for callback interfaces. But we could check for the simple case in https://github.com/jsdom/webidl2js/pull/172/files#diff-7dc541e84f0e601cb83c2704397cb01aR72-R75